src/engine/Sprite.cxx

```diff
@@ -30,7 +30,18 @@ void Sprite::render() const
     {
       if (highlightSprite)
       {
-        SDL_SetTextureColorMod(m_SpriteData[currentLayer].texture, highlightColor.r, highlightColor.g, highlightColor.b);
+        if (SDL_SetTextureColorMod(m_SpriteData[currentLayer].texture, highlightColor.r, highlightColor.g, highlightColor.b))
+        {
+          //TODO indicates failure - an approperiate LOG message should be printed.
```

Comment:
why should this fail?

Either remove this or throw an error

```cpp
    throw CytopiaError(TRACE_INFO "Failed to use SDL_SetTextureColorMod: " + string{SDL_GetError()});
```

but i don't think this is necessary tbh. I'd rather remove the check, since this function is called every frame for each sprite and adds more load to the game